public/calendar_reboot.js

```diff
@@ -918,15 +1052,18 @@ var LocalCalendarStorage = function(params) {
         // Return a promise to the active_calendar id from storage
         
         return new Promise( function(resolve, reject) {
+            //self.startWork() //?
             var activeCalendarId = loadFromLocalStorage(toKey(current_active_calendar));
             
             if (activeCalendarId !== null ) {
                 
                 // Signal that the promise succeeded and make the value ready to go. 
                 jitter(resolve, activeCalendarId);
+                //self.endWork() //?
             }
             else {
                 jitter(reject, "Not found");
+                //self.endwork() //?
```

Comment:
I explained the rational behind when you should or should not call `endWork` below.